Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/Navigation.kt

```diff
@@ -16,28 +16,124 @@
 
 package com.example.compose.jetsurvey
 
-import androidx.fragment.app.Fragment
-import androidx.navigation.fragment.findNavController
-import java.security.InvalidParameterException
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.example.compose.jetsurvey.Destinations.SIGN_IN_ROUTE
+import com.example.compose.jetsurvey.Destinations.SIGN_UP_ROUTE
+import com.example.compose.jetsurvey.Destinations.SURVEY_RESULTS_ROUTE
+import com.example.compose.jetsurvey.Destinations.SURVEY_ROUTE
+import com.example.compose.jetsurvey.Destinations.WELCOME_ROUTE
+import com.example.compose.jetsurvey.signinsignup.SignInScreen
+import com.example.compose.jetsurvey.signinsignup.SignInViewModel
+import com.example.compose.jetsurvey.signinsignup.SignInViewModelFactory
+import com.example.compose.jetsurvey.signinsignup.SignUpScreen
+import com.example.compose.jetsurvey.signinsignup.SignUpViewModel
+import com.example.compose.jetsurvey.signinsignup.SignUpViewModelFactory
+import com.example.compose.jetsurvey.signinsignup.WelcomeScreen
+import com.example.compose.jetsurvey.signinsignup.WelcomeViewModel
+import com.example.compose.jetsurvey.signinsignup.WelcomeViewModelFactory
+import com.example.compose.jetsurvey.survey.PhotoUriManager
+import com.example.compose.jetsurvey.survey.SurveyResultScreen
+import com.example.compose.jetsurvey.survey.SurveyRoute
+import com.example.compose.jetsurvey.survey.SurveyViewModel
+import com.example.compose.jetsurvey.survey.SurveyViewModelFactory
 
-enum class Screen { Welcome, SignUp, SignIn, Survey }
+object Destinations {
+    const val WELCOME_ROUTE = "welcome"
+    const val SIGN_UP_ROUTE = "signup/{email}"
+    const val SIGN_IN_ROUTE = "signin/{email}"
+    const val SURVEY_ROUTE = "survey"
+    const val SURVEY_RESULTS_ROUTE = "surveyresults"
+}
 
-fun Fragment.navigate(to: Screen, from: Screen) {
-    if (to == from) {
-        throw InvalidParameterException("Can't navigate to $to")
-    }
-    when (to) {
-        Screen.Welcome -> {
-            findNavController().navigate(R.id.welcome_fragment)
+@Composable
+fun JetsurveyNavHost(
+    navController: NavHostController = rememberNavController(),
+) {
+    NavHost(
+        navController = navController,
+        startDestination = WELCOME_ROUTE,
+    ) {
+        composable(WELCOME_ROUTE) {
+            val welcomeViewModel: WelcomeViewModel = viewModel(factory = WelcomeViewModelFactory())
+            WelcomeScreen(
+                onSignInSignUp = { email ->
+                    welcomeViewModel.handleContinue(
```

Comment:
Should this logic should be handled within `WelcomeScreen` and `JetsurveyNavHost`? This way, `JetsurveyNavHost` is only concerned with navigation events (i.e. once sign in/up completes and ViewModels would only be scoped within individual screens. Same comment for other screens.

Reply:
That makes sense. I'll create a route composable for each route that has the logic for getting the view model and any setup needed for the individual screen.